engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+
+/**
+ *
+ */
+public class RenderGraph { // TODO: add extends DirectedAcyclicGraph<Node>
+    private List<Node> nodes;
+
+    public RenderGraph() {
+        nodes = Lists.newArrayList();
+    }
+
+    public String add(Node node, String suggestedId) {
+        nodes.add(node);
+        return null; // TODO: for instance if "blur" present make id "blur1" and return it
+    }
+
+    public String add(Node node) {
```

Comment:
I think so. IDs are going to be quite important. A clear render graph with clearly labelled nodes will be very useful for Terasology and modules developers alike, to locate the appropriate position for a new node in the graph or just to do some debugging. 

I mean, if in the logs you find "Node xh35b2k failed to process.", How would you know what that node is about or who introduced it?

If somebody really wants a unique id they can easily generate their own and rely on add(node, id) to return a truly unique one. But I wouldn't encourage the practice.

There could be grounds to somehow make nodes module-unique though, so that module developers can use whatever id they want within the context of the nodes provided by their modules. In this context the id could actually be ModuleId+SuggestedId. 

But let's not worry about this right now, let's just remove the add(node) method for the time being.